src/main/java/io/kestra/plugin/azure/cli/AzCLI.java

```diff
@@ -0,0 +1,154 @@
+                                "  - az ad user list"
+                        }
+                ),
+                @Example(
+                        title = "List all tenant's VMs successfully provisioned using a service principal authentication",
+                        code = {
+                                "username: \"<app-id>\"",
+                                "password: \"secret('az-sp-pass-or-cert')\"",
+                                "tenant: \"<tenant-id>\"",
+                                "servicePrincipal: true",
+                                "commands:",
+                                "  - az vm list --query \"[?provisioningState=='Succeeded']\""
+                        }
+                ),
+                @Example(
+                        title = "Command without authentication",
+                        code = {
+                                "commands:",
+                                "  - az --help"
+                        }
+                )
+        }
+)
+public class AzCLI extends Task implements RunnableTask<ScriptOutput> {
+    @Schema(
+            title = "The commands to run."
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    @NotEmpty
+    private List<String> commands;
+
+    @Schema(
+            title = "Account username."
+    )
+    @PluginProperty(dynamic = true)
+    private String username;
+
+    @Schema(
+            title = "Account password."
+    )
+    @PluginProperty(dynamic = true)
+    private String password;
+
+    @Schema(
+            title = "Tenant id to use."
+    )
+    @PluginProperty(dynamic = true)
+    private String tenant;
+
+    @Schema(
+            title = "Is the account a service principal ?"
+    )
+    @PluginProperty(dynamic = true)
+    private boolean servicePrincipal;
+
+    @Schema(
+            title = "Additional environment variables for the current process."
+    )
+    @PluginProperty(
+            additionalProperties = String.class,
+            dynamic = true
+    )
+    protected Map<String, String> env;
+
+    @Schema(
+            title = "Docker options when for the `DOCKER` runner."
+    )
+    @PluginProperty
+    @Builder.Default
+    protected DockerOptions docker = DockerOptions.builder()
+            .image("mcr.microsoft.com/azure-cli")
+            .build();
+
+    @Override
+    public ScriptOutput run(RunContext runContext) throws Exception {
+        List<String> loginCommands = getLoginCommands();
+
+        CommandsWrapper commands = new CommandsWrapper(runContext)
+                .withWarningOnStdErr(true)
+                .withRunnerType(RunnerType.DOCKER)
+                .withDockerOptions(this.docker)
+                .withCommands(
+                        ScriptService.scriptCommands(
+                                List.of("/bin/sh", "-c"),
+                                loginCommands,
+                                this.commands)
+                );
+
+        commands = commands.withEnv(this.env);
```

Comment:
It is not rendered whereas the property is dynamic

Reply:
The env are already rendered inside the CommandsWrapper